lib/main.py

```diff
@@ -1,35 +1,184 @@
+
+
+@APP.put("/enabled")
+async def enabled_callback(
+    enabled: bool, nc: typing.Annotated[NextcloudApp, Depends(nc_app)]
+):
+    return JSONResponse(content={"error": enabled_handler(enabled, nc)})
+
+
+@APP.api_route(
+    "/{path:path}", methods=["GET", "POST", "PUT", "DELETE", "HEAD", "PATCH", "OPTIONS"]
+)
+async def proxy_Requests(_request: Request, path: str):
+    response = await proxy_request_to_server(_request, path)
+
+    headers = dict(response.headers)
+    headers.pop("transfer-encoding", None)
+    headers.pop("content-encoding", None)
+    headers["content-length"] = str(response.content.__len__())
+    headers["content-security-policy"] = (
+        "default-src * 'unsafe-inline' 'unsafe-eval' data: blob:;"
+    )
+
+    return Response(
+        content=response.content,
+        status_code=response.status_code,
+        headers=headers,
+    )
+
+
+async def proxy_request_to_server(request: Request, path: str):
+    async with httpx.AsyncClient(follow_redirects=False) as client:
+        backend_url = get_backend_url()
+        url = f"{backend_url}/{path}"
+        headers = {}
+        for k, v in request.headers.items():
+            # NOTE:
+            # - remove 'host' to make op routes work
+            # - remove 'origin' to validate csrf
+            if k == "host" or k == "origin":
+                continue
+            headers[k] = v
+
+        if request.method == "GET":
+            params=request.query_params
+            # A referrer header is required when we request to '/work_packages/menu' enpoint
+            # Currently the browser does not provide the referer header so it has been put through proxy
+            # Also it works even referrer is empty
+            if url.endswith("/work_packages/menu"):
+                headers.update({'referer': ''})
+
+            if "/project_storages/new" in url :
+                # when requesting the storate_id is stripped in the proxy (issue: https://github.com/cloud-py-api/app_api/issues/384). 
+                # This piece of code modifies the query param to add missing storage_id.
+                query_params = dict(params)
+                if 'storages_project_storage[]' in query_params:
+                    value = query_params['storages_project_storage[]']
+                    new_key = 'storages_project_storage[storage_id]'
+                    query_params[new_key] = value
+                    del query_params['storages_project_storage[]']
+                params = urlencode(query_params, doseq=True)
+            response = await client.get(
+                url,
+                params=params,
+                headers=headers,
+            )
+        else:
+            # for getting oauth token the content-type 'application/x-www-form-urlencoded' seems to be required
+            if request.method == 'POST' and "/oauth/token" in url:
+                headers.update({'content-type': 'application/x-www-form-urlencoded'})
```

Comment:
what is the conten-type set when the proxy receives the request? Is it also removed?

Reply:
The content type is `application/json` which seems to work for other but for `oauth/token` endpoint the `content-type` `application/json` did not work as `OpenProject` gave `Unsupported Media Type` error while updating it to it worked. And i tried the same thing with postman and the behavior is similar.

Reply:
yes, but how does it work if we run OpenProject without the proxy?

Reply:
I checked it again. It might fixed in `app_api` in newer version as i used 2.8.0 previously and i am using `3.2.0`.
LOGS:
```
http://localhost:3000/stable300/index.php/apps/app_api/proxy/openproject-nextcloud-app/oauth/token
{'content-type': 'application/x-www-form-urlencoded', 'ex-app-id': 'openproject-nextcloud-app', 'ex-app-version': '0.0.1', 'user-agent': 'Nextcloud OpenProject integration', 'accept-encoding': 'gzip', 'x-origin-ip': '127.0.0.1', 'aa-version': '3.2.0', 'authorization-app-api': 'YWRtaW46MTIzNDU=', 'aa-request-id': 'ZvpJ2Uem8QNYdqOIoUdcxAAAAAQ', 'accept-language': 'en', 'content-length': '443'}
```